src/social_network_link_prediction/dimensionality_reduction_methods/nmf.py

```diff
@@ -4,7 +4,49 @@
 from social_network_link_prediction.utils import to_adjacency_matrix
 
 
-def link_prediction_nmf(graph, num_features=2, num_iterations=100):
+def link_prediction_nmf(graph: nx.Graph,
+                        num_features: int = 2,
+                        num_iterations: int = 100) -> csr_matrix:
+    """Compute the _Non-negative Matrix Factorization_ Decomposition for the Graph Adjacency Matrix.
+    The similarity decinoisutuin is defined as:
+
+    .. math::
+        X_\\pm \\approx F_+ G^T_+
+
+    where \\(F \\in \\mathbb{R}^{p \\times k}\\) contains
+    the bases of the latent space and is called the basis matrix;
+    \\(G \\in \\mathbb{R}^{n \\times k}\\) contains combination of coefficients
+    of the bases for reconstructing the matrix \\(X\\), and is called
+    the coefficient matrix; \\(k\\) is the dimention of the latent space
+    ( \\(k<n\\) ) and \\(n\\) is the nunber of data vector
+    (as columns) in \\(X\\).
+
+    Parameters
+    ----------
+    graph: nx.Graph :
+        input Graph (a networkx Graph)
+    num_features: int : # TODO Ricontrollare
+        dimention of the latent space (must be \\(< n\\))
+         (Default value = 2)
+    num_iterations: int : # TODO
+         (Default value = 100)
+
+    Returns
+    -------
+    predicted_adj_matrix: csr_matrix : the Similarity Matrix (in sparse format)
+
+    Notes
+    -----
+    # TODO ricontrollare e capire cosa metterci
```

Comment:
@CristianCosci che descrizione ci mettiamo ? Questa è identica a quella dell'SVD anche perchè fanno praticamente la stessa cosa.

Reply:
Risolto in 64c81b7